hack/clean.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+#
+# Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+echo "> Clean"
+
+for source_tree in $@; do
+  find "$(dirname "$source_tree")" -type f -name "controller-registration.yaml" -exec rm '{}' \;
```

Comment:
This script is used by actual extensions, e.g. [here](https://github.com/gardener/gardener-extension-provider-aws/blob/master/Makefile#L45). I get your point of not having it here, but almost all the scripts in `hack/` are designed to be used from outside extension repos (see https://github.com/gardener/gardener-extension-provider-aws/blob/master/Makefile). Removing it here would mean to add it to every single extension repo. WDYT?

Reply:
Ok, makes sense, that these scripts are designed to be useable from the extensions repos.
But, if I see it correctly, the make goals only pass `./pkg/...` to `clean.sh`, so we wouldn't find a file named `controller-registration.yaml` anyways because they are located under `example`.
I guess, we should adapt the make goals then after merging, if we want `make clean` to delete the `controller-registration.yaml`, WDYT?

Reply:
Yeah, valid point, I remove it from here. Thanks.